tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/PackageRestore/PackageRestoreDataSourceTests.cs

```diff
@@ -10,6 +10,101 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore
 {
+    public class NuGetRestoreServiceTests
+    {
+        [Fact]
+        public async Task NominateAsyncCallsThroughToNuGetNominate()
+        {
+            bool nominateCalled = false;
+
+            var project = UnconfiguredProjectFactory.Create(fullPath: @"C:\Test\Test.csproj");
+            var vsNuGetSolutionRestoreService = IVsSolutionRestoreServiceFactory.ImplementNominateProjectAsync((path, info, ct) => nominateCalled = true);
+            var vsNuGetSolutionRestoreService4 = IVsSolutionRestoreService4Factory.Create();
+            var projectAsynchronousTasksService = IProjectAsynchronousTasksServiceFactory.Create();
+            var restoreService = new NuGetRestoreService(project, vsNuGetSolutionRestoreService, vsNuGetSolutionRestoreService4, projectAsynchronousTasksService);
+
+            var restoreInfo = ProjectRestoreInfoFactory.Create(msbuildProjectExtensionsPath: @"C:\Alpha\Beta");
+            var configuredInputs = PackageRestoreConfiguredInputFactory.Create(restoreInfo);
+
+            var result = await restoreService.NominateAsync(restoreInfo, configuredInputs, default);
+
+            Assert.True(nominateCalled);
+        }
+
+        [Fact]
+        public async Task UpdateDoesNotCallThroughToNuGetNominate()
+        {
+            bool nominateCalled = false;
+
+            var project = UnconfiguredProjectFactory.Create(fullPath: @"C:\Test\Test.csproj");
+            var vsNuGetSolutionRestoreService = IVsSolutionRestoreServiceFactory.ImplementNominateProjectAsync((path, info, ct) => nominateCalled = true);
+            var vsNuGetSolutionRestoreService4 = IVsSolutionRestoreService4Factory.Create();
+            var projectAsynchronousTasksService = IProjectAsynchronousTasksServiceFactory.Create();
+            var restoreService = new NuGetRestoreService(project, vsNuGetSolutionRestoreService, vsNuGetSolutionRestoreService4, projectAsynchronousTasksService);
+
+            var restoreInfo = ProjectRestoreInfoFactory.Create(msbuildProjectExtensionsPath: @"C:\Alpha\Beta");
+            var configuredInputs = PackageRestoreConfiguredInputFactory.Create(restoreInfo);
+
+            await restoreService.UpdateWithoutNominationAsync(configuredInputs);
+
+            Assert.False(nominateCalled);
+        }
+
+        [Fact(Skip = "Cannot ensure that RegisterRestoreInfoSourceAsync has been called when LoadAsync completes.")]
```

Comment:
See earlier comment on a potential way to make these tests deterministic so we can avoid skipping them.

Reply:
Done.